### PR TITLE
Fixed Geant4 Stepping and Stacking actions

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
For the case, when G4GammaGeneralProcess is enabled creator process field in MC truth was incorrectly filled. If stepping verbosity is enabled in forward detectors process limiting step was incorrectly printed.

Both modifications does not affect mainstream MC, because these features are optional. So, no change in results are expected.

#### PR validation:
private

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimG4Core/Application/src/StackingAction.cc [hunk 1]
```diff
@@ -211,6 +212,18 @@ G4ClassificationOfNewTrack StackingAction::ClassifyNewTrack(const G4Track* aTrac
       // potentially good for tracking
       const double ke = aTrack->GetKineticEnergy();
 
+      auto proc = aTrack->GetCreatorProcess();
+      G4int subType = proc->GetProcessSubType();
+      if (nullptr != proc && subType == 16) {
```
Comment (perrotta) [inline SimG4Core/Application/src/StackingAction.cc:217]:
```suggestion
      G4int subType = [any reasonable initialization]
      auto proc = aTrack->GetCreatorProcess();
      if (nullptr != proc) {
        G4int subType = proc->GetProcessSubType();
        if (subType == 16) {
```
if you need to check `proc` not being a null pointer

**Review by perrotta — COMMENTED** [review]
Do I understand correctly that this PR will only affect the LogDebug outputs?

File: SimG4Core/Application/src/StackingAction.cc (hunk 1 above)
Comment (civanch) [reply SimG4Core/Application/src/StackingAction.cc]: @perrotta , thanks, the logic is improved now. For secondary tracks the creator process is always existing, so check is removed. The main update in in line 217-219, which recover MC truth handling for G4GammaGeneralProcess.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
> Do I understand correctly that this PR will only affect the LogDebug outputs?

Ok, with the last commit the process is finally set: https://github.com/cms-sw/cmssw/pull/37779/files#diff-7d6cd9489bf85d971f944e8ae29f23de71b6f37b2183a19b60677640f5b98a93R220
Thank you @civanch !

**Comment by perrotta** [comment]
@smuzaffar comparisons are not accessible from the test summary page. Can they get accessed otherwise? Or should we rerun the tests for it?

**Comment by smuzaffar** [comment]
@perrotta which comparisons results are not accessible? links on https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3db9fd/24425/summary.html work for me

**Comment by perrotta** [comment]
> @perrotta which comparisons results are not accessible? links on https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3db9fd/24425/summary.html work for me

Thank you, they apparently resurrected and I can now access them
(The problem I had was not from the summary itself, but from the comparisons linked from there, now available in https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_4_X_2022-05-03-1100+3db9fd/50024/validateJR.html)

**Comment by perrotta** [comment]
+1

**Pull request merged**